src/utils.js

```diff
@@ -59,18 +61,49 @@ export function isLinkInternal(url) {
 }
 
 // takes in a permalink and returns a URL that links to the image on the staging branch of the repo
-export function prependImageSrc(repoName, chunk) {
-  const $ = cheerio.load(chunk);
+// export function prependImageSrc(repoName, chunk) {
+//   const $ = cheerio.load(chunk);
+//   $('img').each((i, elem) => {
+//     // check for whether the original image source is from within Github or outside of Github
+//     // only modify URL if it's a permalink on the website
+//     if (isLinkInternal($(elem).attr('src'))) {
+//       $(elem).attr('src', `https://raw.githubusercontent.com/isomerpages/${repoName}/staging${$(elem).attr('src')}?raw=true`);
+//     }
+//     // change src to placeholder image if images not found
+//     $(elem).attr('onerror', "this.onerror=null; this.src='/placeholder_no_image.png';")
+//   });
+//   return $.html();
+// }
+
+export async function prependImageSrc(repoName, chunk) {
+  const $ = cheerio.load(chunk)
+  const imagePromises = []
+  const elementsToUpdate = []
+  const queryClient = new QueryClient({
+    defaultOptions: {
+      queries: {
+        staleTime: 1000 * 60 // 60 seconds
+      }
+    }
+  })
+
   $('img').each((i, elem) => {
     // check for whether the original image source is from within Github or outside of Github
     // only modify URL if it's a permalink on the website
     if (isLinkInternal($(elem).attr('src'))) {
-      $(elem).attr('src', `https://raw.githubusercontent.com/isomerpages/${repoName}/staging${$(elem).attr('src')}?raw=true`);
+      const filePath = $(elem).attr('src').substring(1) // remove leading '/'
+      imagePromises.push(queryClient.fetchQuery(filePath, () => fetchImageURL(repoName, filePath)))
+      elementsToUpdate.push(elem)
     }
     // change src to placeholder image if images not found
-    $(elem).attr('onerror', "this.onerror=null; this.src='/placeholder_no_image.png';") 
+    $(elem).attr('onerror', "this.onerror=null; this.src='/placeholder_no_image.png';")
   });
-  return $.html();
+
+  const imageURLs = await Promise.allSettled(imagePromises)
+  for (var i = 0; i<imageURLs.length; i++) {
```

Comment:
We should iterate over the `elementsToUpdate` array using the `.forEach` method so that it is more semantically meaningful.

That is, a reader immediately knows that the intention is "to iterate over the `elementsToUpdate` array", as opposed to "to iterate over an array of equal length to the `elementsToUpdate` array"

(We tend to bias towards using `.map` and `.forEach` array methods as opposed to the general iterator methods because they provide extra context/semantic meaning to the code)